cpp/src/io/orc/dict_enc.cu

```diff
@@ -174,25 +171,22 @@ CUDF_KERNEL void __launch_bounds__(block_size)
 
   if (not dict.is_enabled) { return; }
 
-  auto const t = threadIdx.x;
-  auto map     = map_type::device_view(dict.map_slots.data(),
-                                   dict.map_slots.size(),
-                                   cuco::empty_key{KEY_SENTINEL},
-                                   cuco::empty_value{VALUE_SENTINEL});
-
+  auto t = threadIdx.x;
   __shared__ cuda::atomic<size_type, cuda::thread_scope_block> counter;
 
   using cuda::std::memory_order_relaxed;
   if (t == 0) { new (&counter) cuda::atomic<size_type, cuda::thread_scope_block>{0}; }
   __syncthreads();
-  for (size_type i = 0; i < dict.map_slots.size(); i += block_size) {
-    if (t + i < dict.map_slots.size()) {
-      auto* slot = reinterpret_cast<map_type::value_type*>(map.begin_slot() + t + i);
-      auto key   = slot->first;
+
+  for (; t < dict.map_slots.size(); t += block_size) {
```

Comment:
I prefer adding a local variable here (i.e. `i`) to modifying `t`, which stands for `threadIdx.x` in pretty much all cuIO kernels.